### DISCUSSION TITLE
Transformers with Remix.run

### DISCUSSION THREAD

**Discussion by duesabati** [unocss/unocss#1109] [Q&A]
Hello! First of all thanks for the amazing project! 🌻

I really want to use the variants grouping feature within a remix.run project and I want to do it myself, maybe creating a plugin for unocss. The problem is that really I don't know where to start! I've seen the webpack plugin and saw how variants grouping is implemented in the code, but can someone give me some directions on how it should be done?

Thank you very much in advance!

**Comment by CanRau** [comment]
Just stumbled upon this as well, I'm using unocss as a lib in a resource route but according to [this issue](https://github.com/unocss/unocss/issues/690#issuecomment-1064174133) transformers are currently only supported in Vite?

Also later I realized and probably that's what you're referring to @aliseidev is that transformers require a build integration right?

**Reply by duesabati** [reply]
Hello! Yes I think so, I'm not really sure but if it wasn't like that I'd ask why should it be supported in Vite only? (and then why webpack needed its own plugin).

For my case unfortunately, remix seems like hiding too much the builder (it uses ESBuild the same that powers Vite too) so I think that for now since they don't expose it I can't really do anything. I wish TailwindCSS just implemented it as an opt-in feature.
